lib/firenest/topology.ex

```diff
@@ -215,54 +182,47 @@ defmodule Firenest.Topology do
   or if the operation is not supported.
   """
   @spec disconnect(t, node) :: true | false | :ignored
-  def disconnect(topology, node) do
+  def disconnect(topology, node) when is_atom(topology) and is_atom(node) do
     adapter!(topology).disconnect(topology, node)
   end
 
   @doc """
-  Monitors the given `name` in `node` through `topology`.
-
-  Once the monitored process crashes (or the connection to
-  the node the monitored process runs on is lost), a message
-  with the format described below is delivered to the caller
-  process:
-
-      {:DOWN, ref, :process, {name, node}, reason}
+  Syncs the given `pid` across the topology using its name.
 
-  Where `ref` is the reference returned by `monitor/3` and `reason`
-  is the reason the process exited.
+  This function is the building block for building static services
+  on top of the topology. It allows the current process to know whenever
+  another process with the same name goes up or down in the topology
+  as long as processes call `sync_named/2`.
 
-  This operations always return a `reference`, regardless if
-  there is a connection to `node` or not. In case there is no
-  connection, a DOWN message will be delivered to the caller with
-  reason of `:noconnection`. In case there is no process with `name`
-  in `node`, a DOWN message with reason of `:noproc` is sent.
-  """
-  @spec monitor(t, node, name) :: reference
-  def monitor(topology, node, name) do
-    adapter!(topology).monitor(topology, node, name)
-  end
+  This function returns `{:ok, nodes}` in case the given pid has not
```

Comment:
I'm not sure if I can comment on this PR, but I think you could also document which nodes' names are returned in `nodes` list. The ones where the `name` is already synced?

Reply:
All feedback and comments are welcome. And yes, we need to document this, great catch! And yes, it is the list of nodes that we know are already running name.